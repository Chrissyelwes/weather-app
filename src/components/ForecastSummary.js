import React from "react";
import PropTypes from "prop-types";
import WeatherIcon from "react-icons-weather";

function ForecastSummary(props) {
  const { date, description, icon, temperature } = props;
  const formattedDate = new Date(date).toDateString();

  return (
    <div className="forecast-summary">
      <div className="forecast-summary_date">{formattedDate}</div>
      <div className="forecast-summary_icon" data-testid="forecast-icon">
        <WeatherIcon name="owm" iconId={icon} />
      </div>
      <div className="forecast-summary_temperature">
        {temperature.max}&deg;c
      </div>
      <div className="forecast-summary_description">{description}</div>
      <button type="button">More details</button>
    </div>
  );
}

export default ForecastSummary;

ForecastSummary.propTypes = {
  date: PropTypes.number.isRequired,
  description: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired,
  temperature: PropTypes.shape({
    min: PropTypes.number,
    max: PropTypes.number,
  }).isRequired,
};
