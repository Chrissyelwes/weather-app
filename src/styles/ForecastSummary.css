.forecast-summary {
    font-size: 21px;
    font-family: 'Roboto', sans-serif;
}

.forecast-summary_card {
    display: flex;
    flex-direction: column;
    justify-content: center;
    text-align: center;
    padding: 30px;
    background-color: whitesmoke;
    color: #7E5444;
    border-radius: 5px;
    box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2);
    transition: 0.3s;
    opacity: 80%;
}

.forecast-summary_card:hover {
    box-shadow: 0 8px 16px 0 rgba(0,0,0,0.2);
}

.forecast-summary_date {
    font-size: larger;
    font-weight: bolder;
    color: #A3904C;
    margin-bottom: .625rem;
}


.forecast-summary_temperature {
    font-size: larger;
    font-weight: bolder;
    color: #A3904C;
    margin: .3125rem;
}

.forecast-summary_button {
    padding: 12px 28px;
    font-family: 'Raleway', sans-serif;
    margin-top: .625rem;
    color: whitesmoke;
    background-color: #A3904C;
    border: none;
    border-radius: .3125rem;
    transition-duration: 0.4s;
}

.forecast-summary_button:hover {
    background-color: #7E5444;
}


