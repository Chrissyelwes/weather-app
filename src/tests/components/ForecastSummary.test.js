import React from "react";
import { render } from "@testing-library/react";
import ForecastSummary from "../../components/ForecastSummary";

describe("ForecastSummary", () => {
  const validProps = {
    date: 1111111,
    description: "Stub description",
    icon: "800",
    temperature: {
      min: 12,
      max: 22,
    },
    onSelect: () => {},
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <ForecastSummary
        date={validProps.date}
        description={validProps.description}
        icon={validProps.icon}
        temperature={validProps.temperature}
        onSelect={validProps.onSelect}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correct values for props", () => {
    const { getByText, getByTestId } = render(
      <ForecastSummary
        date={validProps.date}
        description={validProps.description}
        icon={validProps.icon}
        temperature={validProps.temperature}
        onSelect={validProps.onSelect}
      />
    );

    expect(getByText("Thu Jan 01 1970")).toHaveClass("forecast-summary_date");
    expect(getByText("Stub description")).toHaveClass(
      "forecast-summary_description"
    );
    expect(getByTestId("forecast-icon")).toHaveClass("forecast-summary_icon");
    expect(getByText("22ËšC")).toHaveClass("forecast-summary_temperature");
  });
});
