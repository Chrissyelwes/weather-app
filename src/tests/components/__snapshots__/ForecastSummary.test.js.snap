// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummary renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-summary"
    data-testid="forecast-summary"
  >
    <div
      class="forecast-summary_card"
    >
      <div
        class="forecast-summary_date"
      >
        Thu Jan 01 1970
      </div>
      <div
        class="forecast-summary_icon"
        data-testid="forecast-icon"
      >
        <i
          class="wi wi-day-sunny"
          name="owm"
        />
      </div>
      <div
        class="forecast-summary_temperature"
      >
        22ËšC
      </div>
      <div
        class="forecast-summary_description"
      >
        Stub description
      </div>
      <button
        class="forecast-summary_button"
        type="button"
      >
        More details
      </button>
    </div>
  </div>
</DocumentFragment>
`;
